components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx

```diff
@@ -190,6 +192,13 @@ export const ImportAccountModal = () => {
     history.push(WalletRoutes.ImportAccountModal.replace(':accountTypeName?', accountType.name.toLowerCase()))
   }, [])
 
+  React.useEffect(() => {
+    if (hasImportError === false) {
+      history.push(WalletRoutes.Accounts)
+      setImportError(undefined)
```

Comment:
shouldn't we dispatch before navigating?

Reply:
Good call, switched!